ui/src/app/services/workspace-storage.service.ts

```diff
@@ -0,0 +1,77 @@
+import 'rxjs/Rx';
+
+import {Injectable} from '@angular/core';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {Observable} from 'rxjs/Observable';
+import {ReplaySubject} from 'rxjs/ReplaySubject';
+
+
+import {Workspace} from 'generated';
+import {WorkspacesService} from 'generated';
+import {WorkspaceAccessLevel} from 'generated';
+
+export interface WorkspaceData extends Workspace {
+  accessLevel: WorkspaceAccessLevel;
+}
+
+export const BLANK_WORKSPACE: WorkspaceData = {
+  accessLevel: WorkspaceAccessLevel.NOACCESS,
+  name: '',
+  researchPurpose: {
+    diseaseFocusedResearch: false,
+    methodsDevelopment: false,
+    controlSet: false,
+    aggregateAnalysis: false,
+    ancestry: false,
+    commercialPurpose: false,
+    population: false,
+    reviewRequested: false,
+  }
+};
+
+@Injectable()
+export class WorkspaceStorageService {
+  private activeCall = false;
+  private activeWorkspace = new BehaviorSubject<WorkspaceData>(BLANK_WORKSPACE);
```

Comment:
Do all our views support rendering this degenerate blank workspace? UX-wise, do we want to show a flash of empty content on navigate?

I do wonder if we couldn't use this caching approach while keeping the WorkspaceResolver in place, and keep the views dumber (exception: when we submit an update, trigger a reload).

Reply:
Trigger a full page reload after an update? (Just to clarify)

I couldn't find any way to run a refresh of the resolver manually. Only always or never.

Reply:
Sorry - not page reload, I just meant explicitly trigger a refresh of the cached workspace.

Reply:
See @johnstill's comment bullets 3 & 4 - this is a more elegant wording of what I was trying to say on this comment thread.